### PR TITLE
versions file: Also look in /usr/lib/juju

### PR DESCRIPTION
**Pull request by babbageclunk**
## Description of change

For snaps the versions file will be alongside the jujud binary in the
same directory, but when Juju is installed from a deb jujud will be in
/usr/bin, and we won't be able to install the version file there.
Instead, check /usr/lib/juju for the versions file if it's not beside
the binary. (If version files are in both locations the one alongside
the binary takes precedence.) The deb packaging will need to install the
versions file to that directory.

## QA steps

Create a versions file for the locally-built jujud in /usr/lib/juju/jujud-versions.yaml (or move an existing one there).
Bootstrap with an agent-version that isn't in the stream (like 2.3-beta2) and --debug.
See that the version file from /usr/lib/juju is found and included in the agent bundle.
Check that the deployed controller doesn't report a build number in its version.

### CODE REVIEW

File: environs/tools/build_test.go [hunk 1]
```diff
@@ -51,6 +51,11 @@ func (b *buildSuite) SetUpTest(c *gc.C) {
 	dir1 := c.MkDir()
 	dir2 := c.MkDir()
 
+	// Ensure we don't look in the real /usr/lib/juju for jujud-versions.yaml.
+	oldVersionFileFallback := tools.VersionFileFallbackDir
```
Comment (wallyworld) [inline environs/tools/build_test.go]: There's a PatchValue that can be used here - does cleanup automatically

Comment (babbageclunk) [reply environs/tools/build_test.go]: Oh, nice - thanks.

**Comment by babbageclunk** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**